front-end/stateofthecards-front-end/src/Dashboard.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { Component, RefObject, createRef } from "react";
+import styles from "./Dashboard.module.css";
+import MenuCard from "./components/MenuCard";
+import ScaleLoader from "react-spinners/ScaleLoader";
+import QuickJoinMenu from "./components/QuickJoinMenu";
+
+class Dashboard extends Component {
+	// Cards
+	private joinFriendsCard: RefObject<MenuCard> = createRef();
+	private serverListCard: RefObject<MenuCard> = createRef();
+	private createMatchCard: RefObject<MenuCard> = createRef();
+	private quickJoinCard: RefObject<MenuCard> = createRef();
+	private createGameCard: RefObject<MenuCard> = createRef();
```

Comment:
All these `ref`s might be an indication of not using react the way it was intended. Consider using more stateless components and keeping state higher up in a more central component. An even better solution could be `redux` but that's just my opinion